lib/sycamore/sycamore/transforms/base.py

```diff
@@ -297,5 +298,15 @@ def _local_process(self, in_docs: list[Document]) -> list[Document]:
 
         return docs
 
+    def local_execute(self, all_docs: list[Document]) -> list[Document]:
+        docs = [d for d in all_docs if not isinstance(d, MetadataDocument)]
+        metadata = [d for d in all_docs if isinstance(d, MetadataDocument)]
+        outputs = self._local_process(docs)
+        to_docs = [d for d in outputs if not isinstance(d, MetadataDocument)]
+        if self._enable_auto_metadata and (len(docs) > 0 or len(to_docs) > 0):
+            outputs.extend(BaseMapTransform._update_lineage(docs, to_docs))
```

Comment:
This works, but is a bit weird to call the private method on an unrelated class. Please move _update_lineage out of BaseMapTransform.